hack/test-go.sh

```diff
@@ -1,129 +1,250 @@
+    if [[ -n "${test_kube}" ]]; then
+        # we need to find all of the kubernetes test suites, excluding those we directly whitelisted before, the end-to-end suite, and
+        # the go2idl tests which we currently do not support
+        optional_kubernetes_packages="$(find "${kubernetes_path}" -not \(                             \
+          \(                                                                                          \
+            -path "${kubernetes_path}/pkg/api"                                                        \
+            -o -path "${kubernetes_path}/pkg/api/v1"                                                  \
+            -o -path "${kubernetes_path}/test/e2e"                                                    \
+            -o -path "${kubernetes_path}/cmd/libs/go2idl/client-gen/testoutput/testgroup/unversioned" \
+          \) -prune                                                                                   \
+        \) -name '*_test.go' -print0 | xargs -0n1 dirname -z | sort -uz | xargs -0n1 printf "${OS_GO_PACKAGE}/%s\n")"
+
+        test_packages="${test_packages} ${optional_kubernetes_packages}"
+    fi
+fi
 
-  echo 'mode: atomic' > ${OUTPUT_COVERAGE}/profiles.out
-  find $OUTPUT_COVERAGE -name profile.out | xargs sed '/^mode: atomic$/d' >> ${OUTPUT_COVERAGE}/profiles.out
-  go tool cover "-html=${OUTPUT_COVERAGE}/profiles.out" -o "${OUTPUT_COVERAGE}/coverage.html"
+if [[ -n "${dry_run}" ]]; then
+    echo "The following base flags for \`go test\` will be used by $0:"
+    echo "go test ${gotest_flags}"
+    echo "The following packages will be tested by $0:"
+    for package in ${test_packages}; do
+        echo "${package}"
+    done
+    exit 0
+fi
 
-  rm -rf $OUTPUT_COVERAGE/$OS_GO_PACKAGE
+# Run 'go test' with the accumulated arguments and packages:
+if [[ -n "${junit_report}" ]]; then
+    # we need to generate jUnit xml
+    hack/build-go.sh tools/junitreport
+    junitreport="$(os::build::find-binary junitreport)"
+
+    if [[ -z "${junitreport}" ]]; then
+        echo "It looks as if you don't have a compiled junitreport binary"
+        echo
+        echo "If you are running from a clone of the git repo, please run"
+        echo "'./hack/build-go.sh tools/junitreport'."
+        exit 1
+    fi
+
+    test_output_file="${LOG_DIR}/test-go.log"
+    junit_report_file="${ARTIFACT_DIR}/report.xml"
+
+    # we don't care if 'go test' fails, we want to summarize the output anyway.
+    return_code=0 # default to 0, overwrite only if the following line fails
+    go test ${gotest_flags} ${test_packages} > ${test_output_file} 2>&1 || return_code=$?
+    echo "[INFO] Output from \`go test\` redirected to ${test_output_file}"
+
+    cat "${test_output_file}" | "${junitreport}" --type gotest --suites nested --roots github.com/openshift/origin > "${junit_report_file}"
+    echo "[INFO] jUnit XML report placed at ${junit_report_file}"
+    cat "${junit_report_file}" | "${junitreport}" summarize
```

Comment:
Can you use this to clean up the output so it is human readable and streaming in the console?